insta/src/runtime.rs

```diff
@@ -51,43 +55,86 @@ macro_rules! elog {
 #[derive(Debug)]
 pub struct AutoName;
 
-impl From<AutoName> for ReferenceValue<'static> {
-    fn from(_value: AutoName) -> ReferenceValue<'static> {
-        ReferenceValue::File(None)
+pub struct InlineValue<'a>(pub &'a str);
+
+impl From<AutoName> for SnapshotName<'static> {
+    fn from(_value: AutoName) -> SnapshotName<'static> {
+        None
     }
 }
 
-impl From<Option<String>> for ReferenceValue<'static> {
-    fn from(value: Option<String>) -> ReferenceValue<'static> {
-        ReferenceValue::File(value.map(Cow::Owned))
+type SnapshotName<'a> = Option<Cow<'a, str>>;
+
+pub enum SnapshotValue<'a> {
```

Comment:
Can we keep / add the docstrings from https://github.com/mitsuhiko/insta/pull/610/files#diff-ab23e8b51879793793aa0d764a47aa9084b29b94ac7b70e58a9067752ae9e781L87 ?